addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4120,8 +4110,12 @@ registry.ReplaceMedia = SnippetOptionWidget.extend({
      * @override
      */
     async updateUIVisibility() {
-        this.$overlayRemove.toggleClass('d-none', this.$target.is('.fa'));
-        return this._super(...arguments);
+        const _super = this._super(...arguments);
+        // TODO move parent overlay in DOM on top of child one so padding arrows are on top of image border
+        this.$overlay.find('.o_handle').removeClass('d-none');
+        // TODO cleaner way trough `displayHandles` ?
+        this.$overlay.find('.o_inside_parent').addClass('d-none');
```

Comment:
Same question, why a todo instead of a cleaner way then? ^^ But I don't understand the point of the code. Can you explain it? Thanks :+1: 

Is it hiding the overlay? :thinking: Should not we explicitely keep it to handle the part of your commit message "Also, add border around selected image so the used knows what image is selected"

Reply:
> Same question, why a todo instead of a cleaner way then? ^^

Same as above, needed an approval to keep going further in that direction, then I will investigate the `displayHandles` solution -> see [here](https://github.com/odoo-dev/odoo/blame/a82b16ad0b0e74f3ecc632dee5b9ee3e6bcfcfff/addons/web_editor/static/src/js/editor/snippets.editor.js#L503) this attribute is supposed to do the same for `o_inside_parent` element. Somehow it does not if I add the property to the `ReplaceMedia`, but didn't check longer.

> But I don't understand the point of the code. Can you explain it? Thanks 
Is it hiding the overlay?  Should not we explicitely keep it to handle the part of your commit message "Also, add border around selected image so the used knows what image is selected"

It hides the overlay options, see 
![image](https://user-images.githubusercontent.com/30048408/117651057-c6495100-b191-11eb-9c20-a952d205f202.png)

My goal was to re-use the already existing borders that the code is hiding since we show the parent ones.
Doing so, I can fully rely on the already existing code in charge of hiding the overlay/borders when clicked outside.

Might be easier to figure in vidéo:
Before: https://drive.google.com/file/d/1ONTO6vgErwMkiutgk8Vy8eUSIfIRiUMh/view
After: https://drive.google.com/file/d/1G9LjT5lFsDgNvjbL6cFncCAJAkjwiW1X/view

Reply:
Ok, the o_inside_parent class is on the options of the overlay not on the overlay itself, that is why I did not know anymore. But indeed I think the cleaner way would be for that to be the default behavior since we do not want the buttons to be shown for images, seems logic we don't want them ever when there is this double overlay -> the style which comes with o_inside_parent should be replaced by just "display: none"